lib/testutils/src/git.rs

```diff
@@ -350,3 +350,42 @@ impl<'a> IndexManager<'a> {
             .unwrap();
     }
 }
+
+pub fn add_remote(repo: &gix::Repository, remote_name: &str, mut remote: gix::Remote<'_>) {
```

Comment:
nit: I think it's simpler to pass remote url instead of `gix::Remote`. Most callers don't need the remote object.